### ISSUE TITLE
Owncloud client pointing to wrong ssl certificate.

### ISSUE THREAD

**Issue by StrikerASD** [owncloud/client#7123]
### Expected behaviour
After owncloud client is launched, it should log me in into the owncloud server, as always does.

### Actual behaviour
Owncloud launches and there is certificate error, because on apache server i added 3 virtualhosts for owncloud. Those new domain names are added in owncloud config.php file. So the problem is that when i've added more virtualhosts that access the same owncloud instance, my owncloud client asks me about to accept the certificate which newer virtualhost has. This is not a problem on web browsers, because i've checked it. Every domain is working with appropriate certificate.

### Steps to reproduce
1. Start to use owncloud client, make some syncs
2. Configure apache owncloud server for more than one virtualhosts, make ssl certificates for them
3. Check if owncloud website is working fine.
4. Launch owncloud client again and check if you can login. Message about innapropriate certificate should occur.

### Server configuration
Operating system: Debian Stretch 9.8

Web server: Apache 2.4

Database: Mysql

PHP version: 5

ownCloud version: 10.0.10 (stable)

Storage backend (external storage):

### Client configuration
Client version: 2.5.4

Operating system: Windows 10 and Debian Linux 9.8

OS language: PL

**Comment by ckamm** [comment]
@StrikerASD To clarify:
1. You set up owncloud to be accessible on oc1.example.com with a certificate
2. Set up the sync client, accepted the certificate
3. Added new virtual hosts oc2.example.com ... with their own certificates, without changing the oc1.example.com certificate
4. On restart, the client rejects the account certificate and asks you to confirm the certificate again

Is that what happened? What message do you see exactly? Sometimes more details are available in the log file (https://doc.owncloud.org/desktop/2.5/troubleshooting.html#obtaining-the-client-log-file)

**Comment by StrikerASD** [comment]
@ckamm Yes, this is the scenario. Owncloud client (after i switch on two additional virtual hosts) gives me error about certificate of the newer virtualhost. So it should still see oc1.example.com cert, instead he sees oc2.example.com certificate and produces error, because owncloud client still wants to access oc1.example.com. On web browsers this work like a charm, on owncloud client this is the problem. I will send you the log output today or tommorrow.


Regards
StrikerASD

**Comment by ckamm** [comment]
Thanks. I don't understand why the client would even see a difference if it still talks to oc1 and sees the same certificate.

**Comment by StrikerASD** [comment]
Hello,

I've tested owncloud client and it all works well now. I've rebooted the server after update and when it came back again, everything works. No more certificate errors. I've asked my colleague to test it, he replied that on his side, he have certificate errors. When i asked which owncloud version does he have, he answered that this is actually a gnome plugin for ownloud... :)
So for me it's working, i've tested on owncloud clients on Windows and Linux, both working without a hinch now. I don't know what was fixed,because i've had a recent version of apache2 already and i've reloaded apache configuration so the websites were working. Now after an update and reboot everything is fine. Don't have a clue what was wrong. Thank you for your time and sorry for wasting it.

Regards,
Iwo
